scripts/templates/print_helper.py

```diff
@@ -5,103 +5,294 @@
+                self._type_name,
+                "value",
+            )
+        elif re.match("struct", self._obj_type):
+            return PrintArg(
+                " ".join(["const", "struct", self._type_name]),
+                self._type_name,
+                "params",
+            )
+        elif re.match("function", self._obj_type):
+            return PrintArg(
+                " ".join(["const", "struct", self._type_name, "*"]),
+                self._type_name,
+                "params",
+            )
+
+    def _make_type_name(
+        self,
+        namespace: str,
+        tags: Dict[str, str],
+        obj: Dict[str, Union[str, List[Dict[str, str]]]],
+    ) -> str:
+        if re.match("function", self._obj_type):
+            return th.make_pfncb_param_type(namespace, tags, obj)
+        elif re.match("enum", self._obj_type):
+            return th.make_enum_name(namespace, tags, obj)
+        elif re.match("struct", self._obj_type):
+            return th.make_type_name(namespace, tags, obj)
+        return ""
+
+    def _make_c_func_name(self) -> str:
+        if re.match("function", self._obj_type):
+            return th.make_func_name_with_prefix(f"{self._x}Print", self._type_name)
+        elif re.match(r"enum|struct", self._obj_type):
+            return th.make_func_name_with_prefix(f"{self._x}Print", self._name)
+        return ""
+
+    def _make_c_args(self) -> str:
+        c_args = [self.print_arg]
+        c_args.extend(self._c_common_args)
+        return str(c_args).strip("[]")
+
+
+class ExtrasFunc(Func):
+    def __init__(
+        self,
+        x: str,
+        func_name: str,
+        func_args: List[Arg],
+        comment: List[str],
+    ):
+        super().__init__(x)
+        self._func_name = func_name
+        self._func_args = func_args
+        self._comment = comment
+
+    @property
+    def c_name(self) -> str:
+        return self._x + self._func_name[0].capitalize() + self._func_name[1:]
+
+    @property
+    def c_args(self) -> str:
+        return self._make_c_args()
+
+    @property
+    def cpp_name(self) -> str:
+        return self._func_name
+
+    @property
+    def comment(self) -> str:
+        return "".join([line for line in self._comment])
+
+    def _make_c_args(self) -> str:
+        c_args = [
+            arg for arg in self._func_args if not re.match("std::ostream", arg.type)
+        ]
+        c_args.extend(self._c_common_args)
+        return str(c_args).strip("[]")
+
+
+def _arg_from_str(arg_str: str) -> Arg:
+    regex = re.compile(r"((?![&*][&*]+)\w+\Z)")
+    split_arg = re.split(regex, arg_str)
+    arg = Arg(split_arg[0].strip(), split_arg[1])
+    return arg
+
+
+def _add_null_buffer_comment(func_comment: List[str]) -> List[str]:
+    comment_added = False
+    for line in func_comment:
+        if re.search("RESULT_ERROR_INVALID_NULL_POINTER", line):
+            func_comment.append("///         - `NULL == buffer`\n")
```

Comment:
nit: If `RESULT_ERROR_INVALID_NULL_POINTER` is not the last error label, you could append the `NULL == buffer` condition to the wrong error label.